src/frontend/src/lib/components/ui/InputCurrency.svelte

```diff
@@ -8,20 +10,34 @@
 	export let required = true;
 	export let decimals: number | undefined = undefined;
 	export let testId: string | undefined = undefined;
+
+	const dispatch = createEventDispatcher();
+
+	let inputValue: string | undefined;
+	// Convert value to string if it comes from parent (e.g. initial state or set-to-max button)
+	$: inputValue = nonNullish(value) ? `${value}` : undefined;
+
+	const onInput = () => {
+		// Convert inputValue to number before updating parent's value
+		value = nonNullish(inputValue) ? Number(inputValue) : undefined;
+
+		// Bubble nnsInput as consumers might require the event as well
```

Comment:
not really sure we need this comment: I think it is a common pattern in the code of Oisy